core/domain/user_services_test.py

```diff
@@ -1150,3 +1153,179 @@ def test_last_exp_edit_time_gets_updated(self):
         self.assertGreater(
             (owner_settings.last_created_an_exploration),
             previous_last_created_an_exploration)
+
+
+class UserSettingsTests(test_utils.GenericTestBase):
+
+    def setUp(self):
+        super(UserSettingsTests, self).setUp()
+        self.signup(self.OWNER_EMAIL, self.OWNER_USERNAME)
+        self.owner_id = self.get_user_id_from_email(self.OWNER_EMAIL)
+        self.owner = user_services.UserActionsInfo(self.owner_id)
+        self.signup(self.NEW_USER_EMAIL, self.NEW_USER_USERNAME)
+        self.guest_id = self.get_user_id_from_email(self.NEW_USER_EMAIL)
+        user_services.update_user_role(self.guest_id, feconf.ROLE_ID_GUEST)
+
+        self.user_settings = user_services.get_user_settings(self.owner_id)
+        self.user_settings.validate()
+        self.assertEqual(self.owner.role, feconf.ROLE_ID_EXPLORATION_EDITOR)
+
+    def test_validate_non_str_user_id(self):
+        self.user_settings.user_id = 0
+        with self.assertRaisesRegexp(
+            Exception, 'Expected user_id to be a string'):
+            self.user_settings.validate()
+
+    def test_validate_user_id(self):
```

Comment:
Use a test name that is more specific. (Are you validating "nonempty"?)

Reply:
done